### DISCUSSION TITLE
Using UUIDGenerator causes Linker Error (‘CoreAudioTypes' not found)

### DISCUSSION THREAD

**Discussion by nayooti** [pointfreeco/swift-dependencies#125] [General]
### Description

Not sure if this belongs here or is a TCA issue.

Im getting a Linker-Error framework (‘CoreAudioTypes' not found) when importing a framework that uses UUIDGenerator via:
```
@Dependency(\.uuid) var uuid
```

Here is a barebones demo project to demonstrate this issue:
[https://github.com/nayooti/TCADemo/blob/main/DemoFramework/DemoFramework/DemoView.swift](https://github.com/nayooti/TCADemo/tree/main)

Summery:
- DemoApp uses DemoFramework
- DemoFeature does compile
- DemoFramework is a TCA module with a `DemoFeature: Reducer` 
- DemoFeature uses  `@Dependency(\.uuid) var uuid`
- When building DemoApp I get this error:
```
Link DemoFramework (arm64):
  ld: warning: Could not find or use auto-linked framework 'CoreAudioTypes': framework 'CoreAudioTypes' not found
  ld: Undefined symbols:
  Dependencies.Dependency.wrappedValue.getter : A, referenced from:
  DemoFramework.DemoFeature.uuid.getter : Dependencies.UUIDGenerator in DemoView.o
```

Observations:
- Obviously I am not using CoreAudioTypes anywhere in the project
- When commenting out  `@Dependency(\.uuid) var uuid` DemoApp builds without any errors
- Also happens to any other framework that uses DemoFramework

### Checklist

- [ ] I have determined whether this bug is also reproducible in a vanilla SwiftUI project.
- [X] If possible, I've reproduced the issue using the `main` branch of this package.
- [X] This issue hasn't been addressed in an [existing GitHub issue](https://github.com/pointfreeco/swift-dependencies/issues) or [discussion](https://github.com/pointfreeco/swift-dependencies/discussions).

### Xcode version information

Xcode 15.0

**Comment by mbrandonw** [comment]
Hi @nayooti, the project you provided does not have all of the files necessary to compile. It looks like it is missing two projects:

<img width="463" alt="image" src="https://github.com/pointfreeco/swift-dependencies/assets/135203/626a62f6-0bc2-4d26-8963-607b55b7df56">

Can you double check that everything is pushed?

Also, I'm quite certain this problem has nothing to do with this library or TCA. There is a similar issue mentioned in Apple's forums [here](https://developer.apple.com/forums/thread/736152). Most likely you have a linking problem somewhere in your project, and the complaint about "CoreAudioTypes" is just a red herring.

Since I don't think this is an issue with the library I am going to convert it to a discussion. Feel free to follow up with more information.

**Comment by nayooti** [comment]
Hi brandon, thanks for the response.
I updated the project.

**Comment by jshier** [comment]
This is a bug in Xcode 15 / 15.1 and I've encountered it as well. The linking of exported dependencies seems to be broken, so every time a symbol is used from one of TCA's dependencies, it needs to be separately added as a package and explicitly linked.

**Reply by vvisionnn** [reply]
met same issue here, do you find another work around for this issue? seems this only exist when I mark my package as dynamic, when I using default(static), it works as expected

**Comment by nayooti** [comment]
Ok, thank you.
Looks like I will keep my Xcode14-copy a little longer then.
